STIPS/STIP-009.md

```diff
@@ -0,0 +1,240 @@
+
+**StreamingFeeExtension**: Global extension which provides the `owner` and `methodologist` the ability to accrue and split streaming fees at an mutable percentage.
+
+**TradeExtension**: Global extension which provides privileged `operator`(s) the ability to `trade` on a DEX and the `owner` the ability to restrict `operator`(s) permissions with an asset whitelist.
+
+## Requirements
+
+**ManagerFactory**
+
+- Allow `owner` to create new set tokens with a manager smart contract
+- Allow `owner` to migrate existing set tokens to a manager smart contract
+- Allow `owner` to enable extensions and initialize corresponding modules
+- Allow `owner` to create manager smart contract and initialize and parameterize all modules and extensions in two transactions
+
+**BaseManagerV3**
+
+- Allow `owner` to add and remove global `operator` permissions on extensions
+- Allow `owner` to limit `operator` functionality on extensions with an asset whitelist
+- Allow `owner` to update asset whitelist
+- Allow `owner` to perform Set Token admin functions such as `addModule`, `removeModule`, and `setManager`
+
+**BasicIssuanceExtension**
+
+- Allow `owner` to enable functionality of BasicIssuanceModule with only a state change and no contract deployment
+- Allow `operator`(s) to initialize the BasicIssuanceModule
+- Allow users to `issue` and `redeem` the Set Token
+
+**StreamingFeeExtension**
+
+- Allow `owner` to enable functionality of StreamingFeeModule with only a state change and no contract deployment
+- Allow `operator`(s) to initialize the StreamingFeeModule
+- Allow `owner` and `methodologist` to split streaming fees
+- Allow `owner` to update the streaming fee
+- Allow `owner` to update the streaming fee split
+- Allow `owner` to update the streaming fee recipient
+
+**TradeExtension**
+
+- Allow `owner` to enable functionality of TradeModule with only a state change and no contract deployment
+- Allow `operator`(s) to initialize the TradeModule
+- Allow privileged `operator`(s) to perform trades on a DEX
+- Allow `owner` to restrict assets the privileged `operator`(s) can trade into with an asset whitelist
+
+## User Flows
+
+### ManagerFactory.create()
+
+![ManagerFactory create](../assets/stip-009/image2.png "")
+
+An `owner` wants to create a new Set Token with a BaseManagerV3 smart contract manager.
+
+1. The `owner` calls create() passing in parameters to create a Set Token, parameters for the permissioning on BaseManagerV3, and the desired extensions. Specifically,
+
+    - components: List of addresses of components for initial positions
+    - units: List of units for initial positions
+    - name: Name of SetToken
+    - symbol: Symbol of SetToken
+    - owner: The address of the `owner`
+    - methodologist: The address of the `methodologist`
+    - operators: List of addresses of the `operator`(s)
+    - assets: List of addresses of assets for initial asset whitelist
+    - extensions: List of addresses of global extensions to be enabled
```

Comment:
Why are extensions passed in instead of modules? I remember brian mentioning that modules needed to be passed into Set Token creator (it can't be empty). Or are you planning on looking up module address via each extension contract?

Reply:
yes modules should also be passed in